src/com/ref_count.rs

```diff
@@ -1,28 +1,37 @@
-use std::sync::atomic::{self, AtomicU32, Ordering};
+use std::sync::atomic::{self, AtomicI32, Ordering};
 
+/// A thread-safe reference count for use with COM implementations.
 #[repr(transparent)]
 pub struct RefCount {
-    value: AtomicU32,
+    value: AtomicI32,
```

Comment:
Why is this better as an i32? Isn’t it a bug if the count goes below 0?

Reply:
Yes, it is a bug if the count goes below zero. Using `i32` lets us detect this and panic if the object is used after being released. That's the panic below when we release and realize we're already at zero. Hard to test for though...